.github/workflows/getsentry-dispatch.yml

```diff
@@ -8,14 +8,38 @@ on:
   #
   # See https://github.com/getsentry/sentry/pull/21600 for more details
   pull_request_target:
+    types: [labeled, opened, reopened, synchronize]
+
+# disable all other special privileges
+permissions:
+  # needed for `actions/checkout` to clone the code
+  contents: read
+  # needed to remove the pull-request label
+  pull-requests: write
 
 jobs:
   dispatch:
+    if: github.event.action != 'labeled' || github.event.label.name == 'external-trigger-getsentry'
     name: getsentry dispatch
     runs-on: ubuntu-20.04
     steps:
-      # Need to checkout just for `github/file-filters.yml`
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
```

Comment:
this also isn't strictly necessary but helps me sleep better at night -- otherwise `.git/config` will contain a remote with a live token in it (by default this would be a `r/w` token -- though we've scoped it to readonly above)